docs/usage.md

```diff
@@ -1 +1,182 @@
-# Usage
\ No newline at end of file
+# Libvirt-provider - usage documentation
+
+## Overview
+
+`libvirt-provider` enables interaction with `libvirt` for managing virtual machine instances, integrating with multiple plugins for networking, storage, and more. It provides a flexible architecture to handle resources like VMs, volumes, and networking interfaces, with built-in support for garbage collection, volume size resyncing, and health monitoring.
+This guide provides a comprehensive usage flow, including configuration, flags, and practical examples, to get your `libvirt-provider` instance running.
+
+---
+
+## Prerequisites
+
+Before you begin, make sure you have the prerequisites described [here](development/dev_setup.md#prerequisites).
+
+---
+
+## Building from source
+
+To build `libvirt-provider` from the source:
+
+1. **Clone the repository**
+
+    To bring up and start locally the libvirt-provider project, you first need to clone the repository.
+
+    ```bash
+    git clone git@github.com:ironcore-dev/libvirt-provider.git
+    cd libvirt-provider
+    ```
+
+1. **Build the `libvirt-provider`**
+
+    ```bash
+    make build
+    ```
+
+---
+
+## Configuration
+
+### Configuration options
+
+`libvirt-provider` is configured via command-line flags. Below is a description few of the options.
+
+#### General
```

Comment:
Can we maybe drop the list of flags since I see the issue that this will be outdated after some time. 
Maybe you can mention that `libvirt-provider -h` will show the configuration options

Reply:
The hint regarding `--supported-machine-classes` is good, please keep this.